### ISSUE TITLE
Watcher for long computations

### ISSUE THREAD

**Issue by albop** [QuantEcon/QuantEcon.py#125]
I have been using this code on my server to track long computations: https://gist.github.com/albop/8c94e8694369474f7c85 . It is used to track long computations and send an email when they are finished. It uses a context-manager in the following way:
```
with watcher('computation name', email='my@email.com'):
    run_long_computation()
```
When the computation is finished it prints on screen the number of days or seconds elapsed and sends a notification by email.
A few easy improvements could be made to it:
- allow to use desktop notification instead of email
- add an IPython cell/line magic to do the same

What do you think ? If polished a little bit (no docstring), it could be a nice addition.

**Comment by jstac** [comment]
This is neat.  +1 for adding it to the library once the docstring is included, perhaps in `timing.py`.  Alternatively `timing.py` can be renamed `utils.py` as per @spencerlyon2 's suggestion.

**Comment by mmcky** [comment]
Good idea ... +1 for adding it to the library. I have just recently been
thinking of implementing something similar.

**Comment by mmcky** [comment]
@albop I will write and docstring and include this in the new ``utils`` subpackage once the RandomMarkov PR has been merged.  There is a reorganisation of utilities in that branch and I don't want to cause any merge conflicts.

**Comment by mmcky** [comment]
Basic ``tic``, ``tac``, ``tok`` functions are now in ``utils`` sub-package. Leaving open for suggested improvements.

**Comment by albop** [comment]
@mmcky: where does this one stands ? I just realized that I may have forgotten to make the gist public.

**Comment by mmcky** [comment]
Hey @albop - thanks for the reminder. Do you want to submit a PR or should I just add this to the utils subpackage? We will probably want to have an email parameter for the test function - otherwise you may get a lot of test responses :).

**Comment by albop** [comment]
Good catch about the email ! That was careless of me...
I honestly don't know if it's ready to merge. This was more a proof of concept and should probably be tested on a few platforms before merging. I don't know whether it works on windows for instance. Desktop notification would be good too, although not mandatory.

**Comment by mmcky** [comment]
Ok - Cool. I will have a look into this feature next week. I really would like this functionality.
